File: src/api/parentService.js
import axiosClient from './axiosClient';

const parentService = {
  getAll: (page = 0, size = 10) => axiosClient.get('/parent/get-list', { params: { page, size } }),
  getById: (id) => axiosClient.get('/parent/find-by-id', { params: { id } }),
  create: (parentData, role) => axiosClient.post('/parent/create', parentData, { params: { status: role } }),
  update: (id, parentData, role, deleteStatus = 'ACTIVE') => axiosClient.put('/parent/update', parentData, { params: { id, status: role, delete: deleteStatus } }),
  delete: (id) => axiosClient.delete('/parent/delete', { params: { id } }),
};

export default parentService;

File: src/main/java/kg/mega/kindergarten/services/ParentService.java
package kg.mega.kindergarten.services;

import kg.mega.kindergarten.enums.Delete;
import kg.mega.kindergarten.enums.Role;
import kg.mega.kindergarten.models.Parent;
import kg.mega.kindergarten.models.dtos.ParentSaveDto;
import kg.mega.kindergarten.models.dtos.ParentDto;

import java.util.List;

public interface ParentService {
    ParentDto create(ParentSaveDto parentCreateDto, Role role);


    ParentDto delete(Long id);

    List<Parent> findAllList(int page, int size);

    Parent findById(Long id);

    List<Parent> findAll(List<Long> parents);

    ParentDto update(Long id, ParentSaveDto parentSaveDto,Role role, Delete delete);

    List<Object> findChildrenByParentId(Long parentId);
}
